projects/myso-v2/index.js

```diff
@@ -76,10 +72,83 @@ async function tvl(api) {
     ];
   });
 
-  return sumTokens2({
+  return sumTokens2({ api, ownerTokens, });
+};
+
+const getCoveredCallOfTheWeekBalances = async (api, contracts,) => {
+  const coveredCallOfTheWeeks = contracts?.filter(
+    (contract) => contract.type === "p2p"
+  );
+
+  if (!coveredCallOfTheWeeks) return balances;
+
+  const coveredCallOfTheWeekStrategies = await api.multiCall({
+    calls: coveredCallOfTheWeeks.map((contract) => ({
+      target: contract?.contractAddr,
+      params: [contract.poolData.strategyId],
+    })),
+    abi: CoveredCallStrategiesAbi,
+  });
+
+  coveredCallOfTheWeekStrategies.forEach((strategy) => api.add(strategy.underlying, strategy.totalDeposits))
+}
+
+const getWhaleMatchBalances = async (api, contracts, fromBlock) => {
+  const fundingPoolFactory = contracts.find(
+    (contract) => contract.type === "funding_pool_factory"
+  )?.contractAddr;
+
+  const fundingPools =
+    contracts.filter((contract) => contract.type === "funding_pool") || [];
+  console.log(api.chain, fundingPools, fundingPoolFactory)
+
+  if (!(fundingPoolFactory && fundingPools.length))
+    return;
+
+  const logs = await getLogs({
     api,
-    ownerTokens,
+    target: fundingPoolFactory,
+    eventAbi:
+      "event LoanProposalCreated(address indexed loanProposalAddr, address indexed fundingPool, address indexed sender, address collToken, uint256 arrangerFee, uint256 unsubscribeGracePeriod, uint256 numLoanProposals)",
+    onlyArgs: true,
+    fromBlock,
+  });
+
+  if (logs.length) {
+    const loanProposalBalances = await api.multiCall({
```

Comment:
can you explain this part? I tried looking up, but so far there are no loan proposals.

Reply:
We don't have a loan proposal yet, but we will have soon. I wrapped it in try/catch because I couldn't test it properly

Reply:
yes, what is the process here? the tokens from the pool are loan out and some other token is placed as collateral? if so, we can count the loaned out tokens only on under borrowed

Reply:
Currently implemented: Lenders deposit to the funding pool, and after that, lenders can subscribe to available loan proposals (the funding balance decreases and the balance is added to the loan proposal).

Still missing: Execution loan proposal where final collToken amount that borrower needs to transfer is sum of:
- amount reserved for lenders in case of default, and
- amount reserved for lenders in case all convert"


![image](https://github.com/DefiLlama/DefiLlama-Adapters/assets/7297942/d04e7d91-21d4-4272-bd60-bacd8a7d2bfc)